prog/vm/vm_host_slice_nexus.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+class Prog::Vm::VmHostSliceNexus < Prog::Base
+  subject_is :vm_host_slice
+
+  semaphore :destroy, :start_after_host_reboot, :checkup
+
+  def self.assemble_with_host(name, vm_host, family:, allowed_cpus:, memory_gib:, is_shared: false)
+    fail "Must provide a VmHost." if vm_host.nil?
+    fail "Must provide slice name." if name.nil? || name.empty?
+    fail "Must provide family name." if family.nil? || family.empty?
+    fail "Slice name cannot be 'user' or 'system'." if name == "user" || name == "system"
+    fail "Slice name cannot contain a hyphen (-)." if name.include?("-")
+
+    ubid = VmHostSlice.generate_ubid
+
+    DB.transaction do
+      vm_host_slice = VmHostSlice.create(
+        name: name,
+        is_shared: is_shared,
+        family: family,
+        cores: 0,
+        total_cpu_percent: 0,
+        used_cpu_percent: 0,
+        total_memory_gib: memory_gib,
+        used_memory_gib: 0,
+        vm_host_id: vm_host.id
+      ) { _1.id = ubid.to_uuid }
```

Comment:
```suggestion
    DB.transaction do
      vm_host_slice = VmHostSlice.create(
        name: name,
        is_shared: is_shared,
        family: family,
        cores: 0,
        total_cpu_percent: 0,
        used_cpu_percent: 0,
        total_memory_gib: memory_gib,
        used_memory_gib: 0,
        vm_host_id: vm_host.id
      )
```

There's no need to set the id explicitly; it will create the UBID automatically.

https://github.com/ubicloud/ubicloud/pull/2462

Reply:
I remember having some problems with that. Without this code, the id was created in the DB and did not have a format that converted to ubid with the right model class prefix. Can you elaborate how this supposed to work?

Reply:
Or, do I need to use .create_with_id?

Reply:
I will give it a try and see how it works.

Reply:
When did you try? It looks like it works with the latest code.

```ruby
[8] clover-development(main)> VmHostSlice.create(name: "test", cores: 2, total_cpu_percent: 0, used_cpu_percent:0, total_memory_gib: 1, used_memory_gib: 0, family: "standard", vm_host_id: VmHost.first.id)

=> #<VmHostSlice["vsxk5ra9fw2e5kwb6b1bq7da6h"] @values={:name=>"test", :enabled=>false, :is_shared=>false, :cores=>2, :total_cpu_percent=>0, :used_cpu_percent=>0, :total_memory_gib=>1, :used_memory_gib=>0, :family=>"standard", :created_at=>"2025-01-20 23:14:30", :vm_host_id=>"vhjab3390tfe1mt4h2cx53tkmt"}>

[9] clover-development(main)> _.id
=> "eccb8525-fc13-8b79-9f16-6585773b5468"

[10] clover-development(main)> udec "vsxk5ra9fw2e5kwb6b1bq7da6h"
=> #<VmHostSlice["vsxk5ra9fw2e5kwb6b1bq7da6h"] @values={:name=>"test", :enabled=>false, :is_shared=>false, :cores=>2, :total_cpu_percent=>0, :used_cpu_percent=>0, :total_memory_gib=>1, :used_memory_gib=>0, :family=>"standard", :created_at=>"2025-01-20 23:14:30", :vm_host_id=>"vhjab3390tfe1mt4h2cx53tkmt"}>
```

Reply:
Yes, I just pushed the change you suggested. It works. Thank you for the comment. 

It has been a while since this code was first written, and I must had it somewhat different, when it was not working. Anyway, the new version seems fine.

Reply:
Yeah, it merged recently. That’s probably the reason.

https://github.com/ubicloud/ubicloud/pull/2462